internal/app/handlers/batch.go

```diff
@@ -0,0 +1,66 @@
+package handlers
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/KraDM09/shortener/internal/app/config"
+
+	"github.com/KraDM09/shortener/internal/app/models"
+	"github.com/KraDM09/shortener/internal/app/storage"
+	"github.com/KraDM09/shortener/internal/app/util"
+)
+
+type URL struct {
+	CorrelationID string `json:"correlation_id"`
+	Short         string `json:"short_url"`
+}
+
+func BatchHandler(rw http.ResponseWriter, r *http.Request, store storage.Storage) {
+	var req models.BatchRequest
+	dec := json.NewDecoder(r.Body)
+	if err := dec.Decode(&req); err != nil {
+		rw.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	if len(req) == 0 {
+		rw.WriteHeader(http.StatusBadRequest)
+		_, err := rw.Write([]byte("Пустные батчи запрещены"))
+		if err != nil {
+			panic(err)
+		}
+		return
+	}
+
+	resp := make([]URL, len(req))
+	batch := make([]storage.URL, len(req))
+
+	for i, original := range req {
```

Comment:
```suggestion
	for i := range req {
```

а если ты уж решил по индексу обращаться, то делай это до конца.
каждое выделение памяти под original - затратно - используй req[i]

Reply:
принял